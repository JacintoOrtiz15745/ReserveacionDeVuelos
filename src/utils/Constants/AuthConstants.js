const GoogleId = '555459120478-5dvb5sj1o0996s86gg8bk69ogm7vdiu4.apps.googleusercontent.com';  
export const AuthConstants = {
  GoogleId,
};
