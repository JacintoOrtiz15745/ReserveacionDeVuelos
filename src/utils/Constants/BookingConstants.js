const TitleOrigin = 'Where are you now?';
const TitleDestination = 'Where will you be flying to?';
const TextInput = 'Select location';
const BtnText = 'Next';
const SelectDate = 'Select Date';
const Passengers = 'How many passengers?';
const BookingToRoute = 'BookingTo';
const BookingNowRoute = 'BookingNow';
const MyFlightsRoute = 'MyFlights';
const BookingDateRoute = 'BookingDate';
const BookingDoneRoute = 'BookingDone';
const BookingPassengersRoute = 'BookingPassenger';
const doneMessage = 'Your request was received.';
const finish = 'Finish';
const PickerData = [
  {
    label: 'Amsterdam, Netherlands',
    value: 'AMS,Netherlands',
  },
  {
    label: 'Berlin, Germany',
    value: 'BER,Germany',
  },
  {
    label: 'Paris, France',
    value: 'PAR,France',
  },
  {
    label: 'London, UK',
    value: 'LON,UK',
  },
  {
    label: 'Madrid, Spain',
    value: 'MAD,Spain',
  },
];
const passengerNumberData = [
  {
    id: '1',
    title: '1',
  },
  {
    id: '2',
    title: '2',
  },
  {
    id: '3',
    title: '3',
  },
  {
    id: '4',
    title: '4',
  },
  {
    id: '5',
    title: '5',
  },
];
export const BookingConstants = {
  doneMessage,
  TitleOrigin,
  TitleDestination,
  TextInput,
  BtnText,
  SelectDate,
  Passengers,
  BookingToRoute,
  BookingNowRoute,
  MyFlightsRoute,
  passengerNumberData,
  PickerData,
  BookingDateRoute,
  BookingPassengersRoute,
  BookingDoneRoute,
  finish,
};
