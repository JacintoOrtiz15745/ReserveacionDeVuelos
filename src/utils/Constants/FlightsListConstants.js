const planeIconName = 'plane';
const passengers = 'passengers';
const myFlightsTitle = 'My flights';
const flightDataExperimental = [
  {
    id: '1',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'Serbia',
    destinyCity: 'Netherlands',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '2',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '3',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '4',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '5',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '6',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '7',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
];

export const FlightsListConstants = {
  planeIconName,
  passengers,
  myFlightsTitle,
  flightDataExperimental,
  
};
