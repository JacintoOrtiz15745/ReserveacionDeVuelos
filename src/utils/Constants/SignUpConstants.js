const SignUpTitle = 'Sign Up';
const NameInputTitle = 'First Name';
const EmailInputTitle = 'Email *';
const PasswordInputTitle = 'Password *';
const TermsCheckboxTitle = 'I agree to the Terms and Privacy Policy.';
const UpdatesCheckboxTitle = 'Subscribe for select product updates.';
const CharactersMessage =
  'Use 8 or more characters with a mix of letters, numbers, and simbols.';
const SignUpButton = 'Sign Up';
const GoogleSignUpButton = 'Sign Up with Google';
const HasAnAccount = 'Already have an account?';
const LogIn = 'Log In';
const nameErrorMessage = ' must be at least 3 characters.';
const emailErrorMessage = 'invalid Email format';
const passwordErrorMessage = 'must be at least 8 characters';
const OR = 'or';
const SignIn = 'Sign In';
const SignUp = 'Sign Up';
const noAccountMsg = 'DonÂ´t have an account?';
export const SignUpConstants = {
  SignUpTitle,
  NameInputTitle,
  EmailInputTitle,
  PasswordInputTitle,
  TermsCheckboxTitle,
  UpdatesCheckboxTitle,
  CharactersMessage,
  SignUpButton,
  GoogleSignUpButton,
  HasAnAccount,
  LogIn,
  nameErrorMessage,
  emailErrorMessage,
  passwordErrorMessage,
  OR,
  SignIn,
  noAccountMsg,
  SignUp,
};
