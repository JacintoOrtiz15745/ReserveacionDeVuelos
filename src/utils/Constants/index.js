
export * from './SignUpConstants';
export * from './FlightsListConstants'
export * from './AuthConstants'
export * from './BookingConstants'