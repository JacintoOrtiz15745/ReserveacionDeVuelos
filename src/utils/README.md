This folder will contain functions and utilities if needed.