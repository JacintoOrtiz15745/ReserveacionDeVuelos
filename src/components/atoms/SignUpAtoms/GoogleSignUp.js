import React from 'react';
import {Text, TouchableOpacity, Image} from 'react-native';
import {styles} from '../../../styles/SignUpStyles';
import GoogleIcon from '../../../assets/images/google.png';
import {SignUpConstants} from '../../../utils/Constants';
import {signupGoogleFunction} from '../../../../redux/actions/AuthActions';
import {connect} from 'react-redux';
const GoogleSignUp = ({signupGoogleFunction}) => {
  return (
    <TouchableOpacity
      style={styles.btnSignup}
      testID="GoogleSignUp"
      onPress={signupGoogleFunction}>
      <Image
        testID="GoogleIcon"
        source={GoogleIcon}
        style={[styles.googleIcon, {marginLeft: -75}]}
      />
      <Text style={[styles.btnSignupText, {marginLeft: 50}]} testID="btnTitle">
        {SignUpConstants.GoogleSignUpButton}
      </Text>
    </TouchableOpacity>
  );
};
const mapDispatchToProps = {
  signupGoogleFunction,
};
export default connect(null, mapDispatchToProps)(GoogleSignUp);
