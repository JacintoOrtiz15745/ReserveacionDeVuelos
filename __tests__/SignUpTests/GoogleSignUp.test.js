import React from 'react';
import GoogleSignUp from '../../src/components/atoms/SignUpAtoms/GoogleSignUp';
import {shallow} from 'enzyme';
import {SignUpConstants} from '../../src/utils/Constants';
import GoogleIcon from '../../src/assets/images/google.png';
import {connect} from 'react-redux';
import configureMockStore from 'redux-mock-store';

describe('renders correclty', () => {
  let wrapper;
  const mockStore = configureMockStore([]);
  let store;
  store = mockStore({});
  const ConnectedComponent = connect()(GoogleSignUp);

  wrapper = shallow(<ConnectedComponent store={store} />);
  const component = wrapper.dive().dive();
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('component exists', () => {
    const comp = component.findWhere(
      (node) => node.prop('testID') === 'GoogleSignUp',
    );
    expect(comp.exists()).toBeTruthy();
  });
  test('Icon img is displayed with the correct source', () => {
    const img = component.findWhere(
      (node) => node.prop('testID') === 'GoogleIcon',
    );
    expect(img.props().source).toEqual(GoogleIcon);
  });
  test('btn title renders correctly', () => {
    const btnTitle = component.findWhere(
      (node) => node.prop('testID') === 'btnTitle',
    );

    expect(btnTitle.props().children).toEqual(
      SignUpConstants.GoogleSignUpButton,
    );
  });
});
