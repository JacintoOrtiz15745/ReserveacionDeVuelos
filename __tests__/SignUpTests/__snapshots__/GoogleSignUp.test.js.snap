// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correclty snapshot 1`] = `
<Connect(GoogleSignUp)
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
